pkg/operator/latencyprofilecontroller/helpers/helpers.go

```diff
@@ -0,0 +1,102 @@
+package helpers
+
+import (
+	"encoding/json"
+	"fmt"
+	"reflect"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	listersv1 "k8s.io/client-go/listers/core/v1"
+
+	configv1 "github.com/openshift/api/config/v1"
+	"github.com/openshift/library-go/pkg/operator/configobserver"
+	nodeobserver "github.com/openshift/library-go/pkg/operator/configobserver/node"
+	"github.com/openshift/library-go/pkg/operator/latencyprofilecontroller"
+)
+
+const (
+	// static pod revision config maps used by installer controller to track
+	// configs across different revisions
+	revisionConfigMapName = "config"
+	revisionConfigMapKey  = "config.yaml"
+)
+
+// NewInstallerRevisionConfigMatcher is used to create a MatchProfileRevisionConfigsFunc that can be matches
+// config maps generated by installer controller for various static pod revisions and match if each of
+// the active revisions specified contain arg val pairs specific to given latency profile or not.
+func NewInstallerRevisionConfigMatcher(
+	configMapLister listersv1.ConfigMapNamespaceLister,
+	latencyConfigs []nodeobserver.LatencyConfigProfileTuple,
+) latencyprofilecontroller.MatchProfileRevisionConfigsFunc {
+
+	return func(profile configv1.WorkerLatencyProfileType, activeRevisions []int32) (match bool, err error) {
+		// For each revision, check that the configmap for that revision have correct arg val pairs or not
+		for _, revision := range activeRevisions {
+			configMapNameWithRevision := fmt.Sprintf("%s-%d", revisionConfigMapName, revision)
+			configMap, err := configMapLister.Get(configMapNameWithRevision)
+			if err != nil {
+				return false, err
+			}
+
+			match, err := configMatchProfileArguments(configMap, &latencyConfigs, profile)
+			if err != nil {
+				return false, err
+			}
+			// in case a single revision doesn't match return false
+			if !match {
+				return false, nil
+			}
+		}
+		return true, nil
+	}
+}
+
+func configMatchProfileArguments(
+	configMap *corev1.ConfigMap,
+	latencyConfigs *[]nodeobserver.LatencyConfigProfileTuple,
```

Comment:
why do we need a pointer to a slice? Isn't slice enough?

Reply:
A slice would just be fine, I passed pointer to a slice to just save a bit of memory (makes sense? not also sure if it'd), as I guess we're mostly passing around the same contents if `latencyConfigs []nodeobserver.LatencyConfigProfileTuple` here and there, everywhere.

Also, you could assume that `latencyConfigs *[]nodeobserver.LatencyConfigProfileTuple` would be one const-like varibale per operator. So 1 for KCMO, 1 for KAO.

Reply:
Passing a slice is cheap as it is basically a pointer as well

Reply:
Oh cool, I used slice directly now.